docs/docs/adrs/adr-017-allowing-inactive-validators.md

```diff
@@ -0,0 +1,85 @@
+---
+sidebar_position: 18
+title: ADR Template
+---
+# ADR 17: Allowing validators outside the active set to validate on consumer chains
+
+## Changelog
+* 15th May 2024: Initial draft
+
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, only validators in the active set on the provider can validate on consumer chains.
+This limits the number of validators that can validate on consumer chains. Validators outside of the active set might be willing
+to validate on consumer chains, but we might not want to make the provider validator set larger, e.g. to not put more strain on the consensus engine.
+This runs the risk of leaving consumer chains with too few validators.
+
+The purpose of this ADR is to allow validators that are *not* part of the consensus process on the provider chain (because they are inactive)
+to validate on consumer chains.
+
+## Decision
+
+The proposed solution is to:
+* Increase the validator set size in the staking module to a very large number (e.g. 500)
+* Introduce an additional parameter in the provider module, called MaxProviderConsensusValidators
+* When the validator updates are passed to the consensus engine of the provider during EndBlock, the provider module will simply only pass the first MaxProviderConsensusValidators validators by voting power to the consensus engine
+
+To facilitate this, the provider module will need to:
+* keep the last validator set sent to the consensus engine
+* load the current validator set from the staking module
+* diff the two sets, and send the diff to the consensus engine
+
+Extra considerations:
+* Migration: In the migration, the last consensus validator set would just be sent to the last active validator set from the view of the staking module. Existing consumer chains need to be migrated to have a validator set size cap (otherwise, they could end up with a huge validator set including all the staking-but-not-consensus-active validators from the provider chain)
+* Slashing: Validators that are not part of the active set on the provider chain can still be slashed for downtime on a consumer chain, but they *should not* be slashed for downtime on the provider chain. Will those validators accrue missed blocks? If yes, we probably need to make changes in the slashing module to not continuously slash them for downtime on the provider
```

Comment:
> Will those validators accrue missed blocks?

Cf. the [code](https://github.com/cosmos/cosmos-sdk/blob/c059f66ff4264685969a5d16fa55041ed90fb0cd/x/slashing/abci.go#L22) in the slashing module, only the active valset can be slashed for downtime.